index.js

```diff
@@ -57,6 +58,37 @@ module.exports = {
     // called and that is dynamic, so we need to make sure we are requiring the correct parent file everytime.
     delete require.cache[__filename];
 
+    // Finally return the target file's config.
+    return require(targetFile);
+  },
+
+  /**
+   * `delete require.cache()` might not work properly if there is any hook applied to require module, eg: require-in-the-middle
+   * switchV2() will be using stack trace object to get parentFilename and use that to compare with the targetFile.
+   *
+   * It will pick the env config based on the SAILS_ENV value. If a project has SAILS_ENV
+   * defined to some value then it expects that same file in the env/config file so that it
+   * will pick the config value from that file.
+   *
+   * @param defaultConfig this is the default config which is bases on NODE_ENV value
+   * @returns {config}
```

Comment:
For any other reviewers, 
 We're okay with code repetition because eventually, there's going to be one version only.